File: app/src/main/java/com/kjipo/eartraining/NavigationController.java
package com.kjipo.eartraining;


import android.support.v4.app.FragmentManager;

import javax.inject.Inject;

public class NavigationController {
    private final int containerId;
    private final FragmentManager fragmentManager;


    @Inject
    public NavigationController(MainActivity mainActivity) {
        containerId = R.id.frame_container;
        fragmentManager = mainActivity.getSupportFragmentManager();
    }


    public void navigateToNoteView() {
        NoteViewFragment noteViewFragment = new NoteViewFragment();
        fragmentManager.beginTransaction()
                .replace(containerId, noteViewFragment)
                .commitAllowingStateLoss();


    }


}

File: app/src/main/java/com/kjipo/eartraining/NavigationController.kt
package com.kjipo.eartraining


import android.content.Intent
import com.kjipo.eartraining.score.ScoreActivity

class NavigationController constructor(mainActivity: MainActivity) {
    private val mainActivity: MainActivity


    init {
        this.mainActivity = mainActivity
    }

    fun startScoreActivity() {
        val intent = Intent(mainActivity, ScoreActivity::class.java)
        mainActivity.startActivity(intent)
    }


}
